CONTRIBUTING.md

```diff
@@ -1,2 +1,34 @@
+# Contributing
 
-Please have a look at our contributing guidelines at https://github.com/geonetwork/core-geonetwork/wiki/How-to-contribute
+Thank you for contributing to GeoNetwork:
+
+* Free-software: GeoNetwork is free-software, using the [GNU GENERAL PUBLIC LICENSE](LICENSE.md). Contributions provided by you, or your employer, are required to be compatible with this free-software license.
+* Pull-request: GeoNetwork uses a pull-request workflow to review and accept changes. Pull-requests must be submitted against the *master* branch first, and may be back ported as required.
+
+# Pull requests
+
+* Pull request is required, even if you have commit access, so the tests are run and other developer can check your code.
+
+* Pull requests must be applied to `master`, before bing backported.
+
+* Good housekeeping: Anytime you commit, try and clean the code around it to latest style guide. If you improve a function without comments: add comments. If you modify functionality that does not have tests: write a test. If you fix functionality without documentation: add documentation.
+  
+* History: Clean commit messages and history: avoid big commits with hundreds of files, break commits commits up into understandable chunks, longer verbose commit messages are encouraged.
```

Comment:
Note sure if you also want to specify to avoid reformatting code - i.e. indentations..  as it tends to add more code to review than necessary. Also each programmer may have different standard - i.e. 2 spaces for tabs, 4 spaces for tabs, 5 spaces for tabs....  I see that we have a .editorconfig that should help with the unnecessary formatting.

Reply:
I think that is a good matter for discussion, right now I am trying to capture the guidelines from the html docs (even when I think they can be improved).

Reply:
I had a couple moments and resolved this (I agree needless whitespace changes are a challenge)